docs/src/architecture/data-model.md

```diff
@@ -0,0 +1,253 @@
+* The payload contains all fields of the device object in JSON format.
+* Example:
+   * topic: `/tedge/inventory/main`
+   * payload:
+```json
+     {
+        "name": "thin-edge device",
+        "type": "thin-edge.io"   
+     }
+```
+* The next level of the topic structure containes the **capability** objects per device:<br/>
+  `/tedge/inventory/<device id>/<capability type>`
+* Example:
+   * topic: `/tedge/inventory/main/tedge_config`
+   * payload:
+```json
+     {
+        "files": [
+           { "path": "/etc/tedge/tedge.toml", "type": "tedge.toml" },
+           { "path": "/etc/tedge/mosquitto-conf/c8y-bridge.conf" },
+           { "path": "/etc/tedge/mosquitto-conf/tedge-mosquitto.conf" },
+           { "path": "/etc/mosquitto/mosquitto.conf", "type": "mosquitto", "user": "mosquitto", "group": "mosquitto", "mode": "0o644" }
+        ]
+     }
+```
+* All messages to `tedge/inventory` and below are published as retain messages.
+  So one who is interested in any object of the inventory can just subscribe to the object's topic and gets directly the object, if it is available in the inventory.
+
+
+## Registration of a new device
+
+The sequence diagram below illustrates the data/message flow and all components involved, when a new external child-device registers it-self to thin-edge.
+
+```mermaid
+sequenceDiagram
+    participant external child device
+    participant tedge agent
+    participant inventory on MQTT
+    participant C8Y cfg plugin
+    participant C8Y log plugin
+    participant mapper
+    participant C8Y cloud
+        external child device->>tedge agent: (1) register(childid, cfg_capability, log_capability)
+        tedge agent->>inventory on MQTT: (2) create child-device object<br/>and capability objects
+        tedge agent-->>external child device: (3) result    
+
+        inventory on MQTT-->>mapper: (4) notification:<br/>new child-device object(childid)
+        mapper->>C8Y cloud: (5) create child-device twin(childid)
+
+        inventory on MQTT-->>C8Y cfg plugin: (6) notification:new device<br/>object's capability(childid, cfg_capability)
+        C8Y cfg plugin->>mapper: (7) declare operations(<br/>childid, c8y_upload_cfg, c8y_download_cfg)
+        mapper->>C8Y cloud: (8) declare supported operations(<br/>childid, c8y_upload_cfg, c8y_download_cfg, ...)
+        C8Y cfg plugin->>C8Y cloud: (9) declare cfg types(childid, cfg_capability.types)
+        
+        inventory on MQTT-->>C8Y log plugin: (10) notification:new device<br/>object's capability(childid, log_capability)
+        Note right of C8Y log plugin: From here sequence<br>is similar to<br>C8Y cfg plugin        
+```
+
+* Step 1: the external child-device registers to the tedge_agent
+     * Topic:   `tedge/<childid>/commands/req/inventory/register-device`<br/>
+       Payload: **child-device** object with **capability** objects
+     * Example: 
+     
+       Topic: `tedge/child1/commands/req/inventory/register-device`<br/>
```

Comment:
I see some inconsistencies in the topic schemes described here and the topic schemes mentioned in the section above like `tedge/inventory/<device id>` and `/tedge/inventory/<device id>/<capability type>`. As per the previous section, the device registration and capability specific payloads must be sent to separate topics, but this section seems to suggest that they're all sent to the same topic with all the payload combined. Which one is correct?